### PR TITLE
chore(typescript): Migrate ButtonGroup to TypeScript

### PR DESCRIPTION
**Pull request by sahiga**
This PR migrates the `ButtonGroup` component to TypeScript. It also introduces a Puppeteer function for clearing input fields, which ensures that inputs are cleared between snapshots in `test.each` loops.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sahiga has signed the CLA. Thanks for the pull request!

File: src/components/button-group/ButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+
+import './ButtonGroup.scss';
+
+export interface ButtonGroupProps {
+    /** A group of buttons */
+    children: React.ReactNode;
+    /** Class name for ButtonGroup */
+    className?: string;
+    isDisabled?: boolean;
```
Comment (priyajeet) [inline src/components/button-group/ButtonGroup.tsx]: Probably add a doc for this too

Comment (sahiga) [reply src/components/button-group/ButtonGroup.tsx]: @priyajeet Done! Can you review this again?

File: .eslintrc.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ module.exports = {
             files: ['*.test.js', '*.test.tsx'],
             globals: {
                 blurInput: true,
+                clearInput: true,
                 shallow: true,
                 mount: true,
                 takeModalScreenshot: true,
```
Comment (priyajeet) [inline .eslintrc.js]: Just realized, these should be inside storybooks typings file https://github.com/box/box-ui-elements/pull/1890/files#diff-74d358965721573d69d50f8f44457789

Comment (priyajeet) [reply .eslintrc.js]: You should remove it from here btw

Comment (sahiga) [reply .eslintrc.js]: I got a bunch of errors after I removed the globals from this file -- seems like we need them for linting purposes. Do you know of any other solutions?

Comment (priyajeet) [reply .eslintrc.js]: Did you see errors in vscode or when running yarn lint?

Comment (priyajeet) [reply .eslintrc.js]: leave it as is i guess

Comment (sahiga) [reply .eslintrc.js]: I saw errors when running yarn lint

**Pull request merged**